static/js/public/first-snap-flow.js

```diff
@@ -44,60 +43,113 @@ function install(language) {
+    const unselected = document.querySelector(
+      '[class*="js-' + os + '-"]:not(.js-' + type + ")"
+    );
 
     if (!selected && !unselected) {
       return;
     }
 
     if (osWrappers) {
-      osWrappers.forEach(function (wrapper) {
-        const rows = wrapper.querySelectorAll('.js-os-type');
+      osWrappers.forEach(function(wrapper) {
+        const rows = wrapper.querySelectorAll(".js-os-type");
         if (rows) {
-          rows.forEach(function (row) {
-            row.classList.add('u-hide');
+          rows.forEach(function(row) {
+            row.classList.add("u-hide");
           });
         }
       });
     }
 
-    selected.classList.remove('u-hide');
-    unselected.classList.add('u-hide');
+    selected.classList.remove("u-hide");
+    unselected.classList.add("u-hide");
 
-    const continueBtn = document.querySelector('.js-continue');
+    const continueBtn = document.querySelector(".js-continue");
     if (continueBtn) {
       continueBtn.href = `/first-snap/${language}/${type}/package`;
     }
   }
 
-  document.addEventListener('change', onChange);
+  document.addEventListener("change", onChange);
 
-  if (typeof ClipboardJS !== 'undefined') {
-    new ClipboardJS('.js-clipboard-copy');
+  if (typeof ClipboardJS !== "undefined") {
+    new ClipboardJS(".js-clipboard-copy");
   }
 }
 
 function getSnapCount(cb) {
-  fetch('/snaps/api/snap-count').then(r => r.json())
+  fetch("/snaps/api/snap-count")
+    .then(r => r.json())
     .then(data => {
-      cb(data.count);
+      cb(data);
     });
 }
 
+function newArrayValue(arr1, arr2) {
+  let newArr;
+  let oldArr;
+  if (arr1.length === arr2.length) {
+    return false;
+  }
+
+  if (arr1.length > arr2.length) {
+    newArr = arr1;
+    oldArr = arr2;
+  } else {
+    newArr = arr2;
+    oldArr = arr1;
+  }
+
+  let newValues = [];
+
+  newArr.forEach(item => {
+    if (!oldArr.includes(item)) {
+      newValues.push(item);
+    }
+  });
+
+  return newValues;
+}
+
 function push() {
   let initialCount = null;
+  let initialSnaps = [];
   let timer;
   let ready = false;
 
   function getCount(cb) {
     clearTimeout(timer);
 
-    getSnapCount(count => {
+    getSnapCount(data => {
+      if (data.snaps.length !== initialSnaps.length) {
+        const newSnaps = newArrayValue(initialSnaps, data.snaps);
+        if (newSnaps.length > 0 && typeof ga !== "undefined") {
+          ga("gtm1.send", {
+            hitType: "event",
+            eventCategory: "First Snap Flow",
+            eventAction: "Snap pushed",
+            eventLabel: `${newSnaps.join(",")}`
+          });
+        }
+      }
       if (initialCount === null) {
-        initialCount = count;
-      } else if (count !== initialCount) {
+        initialCount = data.count;
+        initialSnaps = data.snaps;
+      } else if (data.count !== initialCount) {
+        const newSnaps = newArrayValue(initialSnaps, data.snaps);
+
+        if (newSnaps.length > 0 && typeof ga !== "undefined") {
```

Comment:
Why is the same condition here as in the line 129? Seems like we send same ga event twice?

Reply:
Removed, was for my testing